### PR TITLE
Add callback to sync space with conditional to stall job completion

### PR DESCRIPTION
**Pull request by jasonlgong**
Why this PR?
- The api call to sync space does not complete before the success notification resulting in poor UX on the HCM show app where records are not upserted in time and not displayed in the respective resources page

### CODE REVIEW

File: workflows/dynamic/index.ts [hunk 1]
```diff
@@ -237,9 +239,11 @@ export default function (listener) {
           // Perform error handling, such as displaying an error message to the user or triggering a fallback behavior
         }
 
-        await api.jobs.complete(jobId, {
-          info: 'Data synced to the HCM.show app.',
-        });
+        if (callback === '{"success":true}') {
```
Comment (jasonlgong) [inline workflows/dynamic/index.ts]: Got stuck on this and not sure why this a string version of the object is necessary. 

SS below of the log for callback

CleanShot 2023-07-18 at 14 23 09@2x

Comment (mhpjay422) [reply workflows/dynamic/index.ts:242]: This does not work as the conditional

```
if (callback && callback.success === true) {
          await api.jobs.complete(jobId, {
            info: 'Data synced to the HCM.show app.',
          });
        }
```

Comment (jasonlgong) [reply workflows/dynamic/index.ts]: Sooo I see now the backend serializes the object into a JSON string. 😞

Comment (jasonlgong) [reply workflows/dynamic/index.ts]: Editing now to parse it, so it doesnt look so ugly

**Review by ryanlabouve — APPROVED** [review]

**Pull request merged**